### PR TITLE
Enhancement: Implement Bootstrap API

### PR DESCRIPTION
**Pull request by gavinmcdermott**
Implement the bootstrap API in reference to issue #217 raised by @diasdavid

### CODE REVIEW

File: test/api/bootstrap.spec.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        expect(peers).to.exist
+        expect(peers.length).to.above(1)
+        done()
+      })
+    })
+  })
+
+  describe('.list', () => {
+    it('returns a list of peers', (done) => {
+      return apiClients.a.bootstrap.list(null, (err, res) => {
```
Comment (diasdavid) [inline test/api/bootstrap.spec.js]: Why do we need the `null` arg?

File: test/api/bootstrap.spec.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  describe('.rm', () => {
+    it('returns an error when called with an invalid arg', (done) => {
+      return apiClients.a.bootstrap.rm(invalidArg, (err) => {
+        expect(err).to.be.an.instanceof(Error)
+        done()
+      })
+    })
+
+    it('returns empty list when called without an arg or options', (done) => {
+      return apiClients.a.bootstrap.rm(null, (err, res) => {
```
Comment (diasdavid) [inline test/api/bootstrap.spec.js]: Does this mean 'delete all'?

**Comment by diasdavid** [comment]
Awesome work @gavinmcdermott ! :D left a couple of questions

File: test/api/bootstrap.spec.js (hunk 1 above)
Comment (gavinmcdermott) [inline test/api/bootstrap.spec.js]: Great question.....we don't

File: test/api/bootstrap.spec.js (hunk 2 above)
Comment (gavinmcdermott) [inline test/api/bootstrap.spec.js]: Nope. According to the spec there is an explicit 'delete all' call—you just have to pass an `all: true` option in order to remove all bootstrap peers. The returned array contains the list of peers that were removed in the call (I'll also make the `it` string a better description).

File: test/api/bootstrap.spec.js (hunk 1 above)
Comment (diasdavid) [inline test/api/bootstrap.spec.js]: Can we take it off then?

Comment (gavinmcdermott) [inline test/api/bootstrap.spec.js]: Of course! Was away from my machine yesterday - should have mentioned in the note that I'd update the pr within the day. It's up now :)

**Comment by diasdavid** [comment]
LGTM :) Thank you @gavinmcdermott 

Could you squash the commits for the merge?

File: test/api/bootstrap.spec.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    })
+  })
+
+  describe('.promise', () => {
+    describe('.add', () => {
+      it('returns an error when called without args or options', (done) => {
+        return apiClients.a.bootstrap.add(null)
+          .catch((err) => {
+            expect(err).to.be.an.instanceof(Error)
+            done()
```
Comment (dignifiedquire) [inline test/api/bootstrap.spec.js]: You can drop the `done` in all the promise tests, mocha detects that you return a promise (also need to remove it from the function signature.

**Pull request merged**